tests/pytorch/test_pin_memory.py

```diff
@@ -0,0 +1,38 @@
+import backend as F
+import dgl
+import unittest
+import torch
+
+@unittest.skipIf(F._default_context_str == 'cpu', reason="Need gpu for this test")
+def test_pin_noncontiguous():
+    t = torch.empty([10, 100]).transpose(0, 1)
+
+    assert not t.is_contiguous()
+    assert not F.is_pinned(t)
+
+    try:
+        dgl.utils.pin_memory_inplace(t) 
+    except dgl.DGLError as e:
```

Comment:
We can use `with pytest.raises(DGLError):` context here.

Reply:
Switched in 6961d66